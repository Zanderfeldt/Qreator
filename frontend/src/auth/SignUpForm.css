/* SignUpForm.css */

.signup-card {
  background-color: #ffffff;
  border-radius: 10px;
  border: solid rgba(39, 35, 33, 0.192) 0.5px;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.295);
  width: 320px;
  margin: 0 auto;
  margin-top: 20px;
  padding: 20px;
  text-align: center;
  font-family: Arial, sans-serif;
}

.signup-card h2 {
  color: #333;
  font-size: 24px;
  margin-bottom: 10px;
}


.signup-card input {
  width: 80%;
  padding: 2px;
  margin: 8px 0;
  border: 1px solid #ccc;
  border-radius: 4px;
  font-size: 14px;
  height: 25px;
}

.signup-card button {
  background-color: #438b49;
  color: #fff;
  padding: 10px 20px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-size: 16px;
  margin-top: 10px;
}

.signup-card button:hover {
  background-color: #70dd5a;
}
