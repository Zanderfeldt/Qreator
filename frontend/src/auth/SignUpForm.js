import React, { useState } from 'react';
import { useHistory } from "react-router-dom";
import './SignUpForm.css';

function SignUpForm({ register }) {
  const INITIAL_STATE = {
    username: '',
    password: '',
    firstName: '',
    lastName: '',
    email: '',
  };

  const [formData, setFormData] = useState(INITIAL_STATE);
  const [showAlert, setShowAlert] = useState('');
  const [errors, setErrors] = useState(INITIAL_STATE);
  const history = useHistory();

  const handleSubmit = async (e) => {
    e.preventDefault();
    // Check if any of the fields are empty, and set the error messages
    const newErrors = { ...INITIAL_STATE };
    if (formData.username.trim() === '') {
      newErrors.username = 'Username is required';
    }
    if (formData.password.trim() === '') {
      newErrors.password = 'Password is required';
    }
    if (formData.firstName.trim() === '') {
      newErrors.firstName = 'First Name is required';
    }
    if (formData.lastName.trim() === '') {
      newErrors.lastName = 'Last Name is required';
    }
    if (formData.email.trim() === '') {
      newErrors.email = 'E-mail is required';
    }
    setErrors(newErrors);

    // If errors, don't proceed with registration
    if (Object.values(newErrors).some((error) => error !== '')) {
      return;
    }

    let result = await register(formData);
    if (result.success) {
      setFormData(INITIAL_STATE);
      history.push('/');
    } else {
      setShowAlert(result.message);
    }
  };

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));

    // Clear the error message when the user starts typing
    setErrors((prevErrors) => ({
      ...prevErrors,
      [name]: '',
    }));
  };

  const isFormValid = Object.values(errors).every((error) => error === '');

  return (
    <div className="signup-card">
      <h2>Sign Up</h2>
      <small style={{ color: 'red' }}>{showAlert}</small>
      <form onSubmit={handleSubmit}>
        <input
          id="username"
          name="username"
          placeholder="Username"
          value={formData.username}
          onChange={handleChange}
        />
        {errors.username && <div className="error-message">{errors.username}</div>}
        <input
          type="password"
          id="password"
          name="password"
          placeholder="Password"
          value={formData.password}
          onChange={handleChange}
        />
        {errors.password && <div className="error-message">{errors.password}</div>}
        <input
          id="firstName"
          name="firstName"
          placeholder="First Name"
          value={formData.firstName}
          onChange={handleChange}
        />
        {errors.firstName && <div className="error-message">{errors.firstName}</div>}
        <input
          id="lastName"
          name="lastName"
          placeholder="Last Name"
          value={formData.lastName}
          onChange={handleChange}
        />
        {errors.lastName && <div className="error-message">{errors.lastName}</div>}
        <input
          id="email"
          name="email"
          placeholder="E-mail"
          value={formData.email}
          onChange={handleChange}
        />
        {errors.email && <div className="error-message">{errors.email}</div>}
        <button disabled={!isFormValid}>Submit</button>
      </form>
    </div>
  );
}

export default SignUpForm;
