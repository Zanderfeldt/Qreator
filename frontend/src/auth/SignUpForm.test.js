import React from 'react';
import { render, screen } from '@testing-library/react';
import SignUpForm from './SignUpForm';

describe('SignUpForm', () => {
  // Smoke Test
  it('should render without errors', () => {
    render(<SignUpForm register={() => {}} />);
  });

  // Snapshot Test
  it('should match the snapshot', () => {
    const { asFragment } = render(<SignUpForm register={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
