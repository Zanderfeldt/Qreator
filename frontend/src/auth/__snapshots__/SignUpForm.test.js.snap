// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="signup-card"
  >
    <h2>
      Sign Up
    </h2>
    <small
      style="color: red;"
    />
    <form>
      <input
        id="username"
        name="username"
        placeholder="Username"
        value=""
      />
      <input
        id="password"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
      <input
        id="firstName"
        name="firstName"
        placeholder="First Name"
        value=""
      />
      <input
        id="lastName"
        name="lastName"
        placeholder="Last Name"
        value=""
      />
      <input
        id="email"
        name="email"
        placeholder="E-mail"
        value=""
      />
      <button
        disabled=""
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
